src/pages/ImportData.js

```diff
@@ -77,40 +77,224 @@ const ResponseFragment = ({ title, snapshotResponses, responseIndex }) => {
+  div({ style: { marginTop: '1rem' } }, [
+    `The ${isDataset ? 'dataset' : 'snapshot'}(s) you just chose to import to Terra will be made available to you `,
+    'within a workspace of your choice where you can then perform analysis.'
+  ])
+])
+
+// ImportDataDestination handles selecting which workspace to import to
+const ImportDataDestination = ({ workspaceId, templateWorkspaces, template, userHasBillingProjects, isDataset, authorizationDomain, onImport, isImporting }) => {
   const { workspaces, refresh: refreshWorkspaces, loading: loadingWorkspaces } = useWorkspaces()
-  const [isImporting, setIsImporting] = useState(false)
-  const { query: { url, format, ad, wid, template, snapshotId, snapshotName, snapshotIds, referrer, tdrmanifest, catalogDatasetId, tdrSyncPermissions } } = Nav.useRoute()
-  const [mode, setMode] = useState(wid ? 'existing' : undefined)
+  const [mode, setMode] = useState(workspaceId ? 'existing' : undefined)
   const [isCreateOpen, setIsCreateOpen] = useState(false)
   const [isCloneOpen, setIsCloneOpen] = useState(false)
-  const [selectedWorkspaceId, setSelectedWorkspaceId] = useState(wid)
   const [selectedTemplateWorkspaceKey, setSelectedTemplateWorkspaceKey] = useState()
-  const [allTemplates, setAllTemplates] = useState()
+
+
+  const [selectedWorkspaceId, setSelectedWorkspaceId] = useState(workspaceId)
+
+  const selectedWorkspace = _.find({ workspace: { workspaceId: selectedWorkspaceId } }, workspaces)
+
+  const filteredTemplates =
+    _.flow(
+      _.flatMap(id => (templateWorkspaces && templateWorkspaces[id]) || []),
+      _.filter(({ name, namespace }) => _.some({ workspace: { namespace, name } }, workspaces))
+    )(_.castArray(template))
+
+  const noteMessage = 'Note that the import process may take some time after you are redirected into your destination workspace.'
+
+  const linkAccountPrompt = () => {
+    return div({}, [
+      div({ style: { marginTop: '1.5rem' } }, ['But first, to use Terra, you need to link Terra to a cloud account for compute and storage costs']),
+      h(ButtonPrimary, { style: { marginTop: '.5rem', padding: '.75rem 3.5rem' }, href: '/billing' }, 'Get Started')
+    ])
+  }
+
+  const SelectExistingWorkspace = () => h(Fragment, [
+    h2({ style: styles.title }, ['Start with an existing workspace']),
+    h(IdContainer, [id => h(Fragment, [
+      h(FormLabel, { htmlFor: id, style: { marginBottom: '0.25rem' } }, ['Select one of your workspaces']),
+      h(WorkspaceSelector, {
+        id,
+        workspaces: _.filter(ws => {
+          return Utils.canWrite(ws.accessLevel) &&
+            (!authorizationDomain || _.some({ membersGroupName: authorizationDomain }, ws.workspace.authorizationDomain))
+        }, workspaces),
+        value: selectedWorkspaceId,
+        onChange: setSelectedWorkspaceId
+      })
+    ])]),
+    isDataset && div({ style: { marginTop: '0.5rem', lineHeight: '1.5' } }, [noteMessage]),
```

Comment:
The only use of `isDataset` in this method is to gate the display of `noteMessage`. If the caller passed in `noteMessage` instead then I think `isDataset` could be removed, and this could be written as:
```suggestion
    noteMessage && div({ style: { marginTop: '0.5rem', lineHeight: '1.5' } }, [noteMessage]),
```

which to me makes it clearer that the test is correct